### DISCUSSION TITLE
Next Steps For Widgets

### DISCUSSION THREAD

**Discussion by jdunkerley** [enso-org/enso#6652] [Work In Progress - GUI]
Summarizing the discussion we had on 10th May.

- Get @Frizi's PR completed and merged as soon as possible to test out more fully.
- Constructor Dropdowns: 
  - [x] Engine team are improving support for ascribed types on the fields and dynamic data - #6584
  - [x] Language Server to be amended to not send all fields as `Any` - #6655
- Control over when to be a List Editor or not:
  - Currently based on if the type contains "Vector".
  - [x] As a quick initial fix will only be a List Editor if the type list starts with `Vector `.
  - Longer term should be able to set the type of a widget using the dynamic data.
- Nested Dropdowns
  - In order to make `cross_tab` a nice experience want to be able to specify the drop down for the Aggregate_Column from the Table.
  - The widget API provides this nested data within the JSON passed back.
  - We agreed to only do one nesting level
  - [x] The dropdown for nested constructors should come from the parent call is supplied. This should work in List Editor drop downs too.
  - Longer term was suggested to make changes to the language server protocol so it could know more of the context when evaluating.
  - This could allow knowing the parent value and also previous parameter values, which could enable more powerful widget experiences.

cc: @farmaazon @sylwiabr @Frizi @wdanilo

**Comment by JaroslavTulach** [comment]
> Control over when to be a List Editor or not:
> Currently based on if the type contains "Vector".

How can a "vector editor" work on anything else than `Vector`? What other type it can work on?

> ... if the type list starts with Vector .

When the "vector editor" can only work on `Standard.Base.Data.Vector.Vector` - why do we check for _"type list starts with Vector"_?

**Reply by farmaazon** [reply]
Yes, it can work on `Standard.Base.Text | Standard.Base.Data.Vector.Vector` type. We want to display it only when the Vector is mentioned first.
